### DISCUSSION TITLE
Monaco Documentation of Output

### DISCUSSION THREAD

**Discussion by ghost** [Dynatrace/dynatrace-configuration-as-code#193] [Archive]
**Is your feature request related to a problem? Please describe.**
Monaco structure is logic for users but not immediately digestable to non-technical users.

**Describe the solution you'd like**
Given a set of monaco configuration projects, monaco should be able to self produce a documentation listing what configuration will be applied.

For example:

Your monaco folder contains 2 projects:
- ProjectA
- ProjectB

`ProjectA` contains the following configuration:

- 1x Auto tag rule called `myTagRule`
- 1x Management zone called `myZone`
- 1x Alerting profile called `alertProfile`
- 1x Problem notification called `alert` which references the `alertProfile` alerting profile.

**Describe alternatives you've considered**
N/A

**Additional context**
The output format of this tool (PDF, Word doc, Markdown) is up for discussion.
The format of the output (text heavy or more visual with linked graphics is up for discussion.

@163a as discussed.

**Comment by UnseenWizzard** [comment]
Do understand correctly that this request is to generate more verbose output of what is being applied? 

Could this be something that would generally make sense to add - in addition to the current simple logging of which projects are being applied/validated and the full verbose log of deployments?

**Reply by ghost** [reply]
Not really, this is more a requirement to pretty print what monaco will apply. The intended audience for this output would be business / non-tech folk who don't know how to interpret the monaco folder structure and linkages.

For 2 examples:
- Sprint retro's: Show what config has been applied
- Explanations to business: Show what config has or will be applied

Really this is just a pretty way to generate a nice looking document given a set monaco project.

In other words, if I hand you a monaco folder structure, the work is on you to figure out what's going on. If I hand you the monaco folder structure + document which explains what hte folder will apply, this is much more user friendly.

`monaco plan`?

**Reply by patrickpichler** [reply]
A plan like terraform/pulumi would be awesome, but I think the 
scope here should really be limited to simple ascii outputs. 

If you want to create fancy graphs/documents with it, I guess
you would then use a different tool which can handle monaco
plan output.

Actually isn't this kinda was the requested `diff`feature does?

**Reply by tatomir146** [reply]
Agree with @patrickpichler. As monaco is cli based tool and not really intended for non-tech folk I wouldn't start implementing fancy reports, but if we need to log more information we could think about adding it. @patrickpichler yes this looks like #202. @Dynatrace-Adam-Gardner would you agree to proceed discussion in #202?
